src/sentry/api/endpoints/organization_spans_aggregation.py

```diff
@@ -0,0 +1,230 @@
+EventSpan = namedtuple(
+    "EventSpan",
+    [
+        "span_id",
+        "is_segment",
+        "parent_span_id",
+        "group",
+        "group_raw",
+        "description",
+        "start_ms",
+        "duration",
+        "exclusive_time",
+    ],
+)
+
+
+class EventSpans(TypedDict):
+    transaction_id: str
+    spans: List[EventSpan]
+
+
+NULL_GROUP = "00"
+
+
+@region_silo_endpoint
+class OrganizationSpansAggregationEndpoint(OrganizationEventsEndpointBase):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+
+    aggregated_tree: Dict[str, Dict[str, Union[str, int, float]]] = {}
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        if not features.has("organizations:starfish-view", organization, actor=request.user):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response(status=404)
+
+        transaction = request.query_params.get("transaction", None)
+        if transaction is None:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST, data={"details": "Transaction not provided"}
+            )
+
+        builder = QueryBuilder(
+            dataset=Dataset.SpansIndexed,
+            params=params,
+            selected_columns=["transaction_id", "count()"],
+            query=f"transaction:{transaction}",
+            limit=100,
+        )
+        builder.columns.append(
+            Function(
+                "groupArray",
+                parameters=[
+                    Function(
+                        "tuple",
+                        parameters=[
+                            Column("span_id"),
+                            Column("is_segment"),
+                            Column("parent_span_id"),
+                            Column("group"),
+                            Column("group_raw"),
+                            Column("description"),
+                            Column("start_ms"),
+                            Column("duration"),
+                            Column("exclusive_time"),
+                        ],
+                    )
+                ],
+                alias="spans",
+            )
+        )
+        snql_query = builder.get_snql_query()
+        snql_query.tenant_ids = {"organization_id": organization.id}
+        results = raw_snql_query(snql_query, Referrer.API_ORGANIZATION_SPANS_AGGREGATION.value)
+        for event in results["data"]:
+            span_tree = {}
+            root_span_id = None
+            spans = event["spans"]
+
+            for span in spans:
+                span = EventSpan(*span)
+                span_id = getattr(span, "span_id")
+                is_root = getattr(span, "is_segment")
+                if is_root:
+                    root_span_id = span_id
+                if span_id not in span_tree:
+                    spans_dict = span._asdict()
```

Comment:
To check, this is the [namedtuple._asdict](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._asdict) right? (Dang really wish they had named this differently)

Reply:
yeah, it's the `EventSpan` named tuple from above